### PR TITLE
feat(data-table): ng7 data-table component

### PR DESCRIPTION
**Pull request by daanishnasir**
## Description

Data Table, Sorting, Column Resizing 

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: angular/src/lib/data-table/data-table.component.ts [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+  /** @prop set the column'n header and field data */
+  @Input() get columns(): any[] {
+    return this._columns;
+  }
+  set columns(cols: any[]) {
+    this._columns = cols;
+    this.tableService.onColumnsChange(cols);
+  }
+
+  /** @prop number of records in data set */
```
Comment (pauljeter) [inline angular/src/lib/data-table/data-table.component.ts]: What is the default for this prop?

File: angular/src/lib/data-table/data-table.component.ts [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+  // hide resize helper line on end
+  this.resizeHelperViewChild.nativeElement.style.display = 'none';
+  }
+}
+
+
+// Body Component
+
+@Component({
+  selector: '[bodyColumns]',
```
Comment (pauljeter) [inline angular/src/lib/data-table/data-table.component.ts]: Does this need the `md` prefix?

File: angular/src/lib/data-table/data-table.component.ts [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+      field: this.field
+    });
+  }
+}
+
+
+// Resizable Column
+
+
+@Directive({
```
Comment (pauljeter) [inline angular/src/lib/data-table/data-table.component.ts]: Should we break these directives(`md-sort-column`, `md-resizable-column`, etc.) out into separate files? This one is getting long.

Comment (daanishnasir) [reply angular/src/lib/data-table/data-table.component.ts]: I agree, its become lengthy, I'll break them into separate files

File: core/scss/components/data-table/data-table.scss [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        width: 1px;
+        position: absolute;
+        z-index: 10;
+        display: none;
+      }
+    }
+
+    //gray even background color rows
+    .md-data-table__tbody > tr:nth-child(even) {
+      // background-color: $md-gray-20;  // zebra
```
Comment (mathsusan) [inline core/scss/components/data-table/data-table.scss]: remove commented code

File: angular/src/lib/data-table/tests/data-table.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      imports: [DataTableModule],
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    // tslint:disable-next-line: no-use-before-declare
+    fixture = TestBed.createComponent(TestAppComponent);
+    component = fixture.componentInstance;
+    // fixture.detectChanges();
```
Comment (mathsusan) [inline angular/src/lib/data-table/tests/data-table.component.spec.ts]: remove commented code

File: angular/src/lib/data-table/resize-column.directive.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+    this.zone.runOutsideAngular(() => {
+      this.mouseDownListener = this.onMouseDown.bind(this);
+      this.resizeHelper.addEventListener('mousedown', this.mouseDownListener);
+    });
+
+  }
+
+  bindDocEvents() {
+    this.zone.runOutsideAngular(() => {
```
Comment (mathsusan) [inline angular/src/lib/data-table/resize-column.directive.ts]: this zone thing is interesting .. very cool

File: angular/src/docs/app/playground/playground.component.ts [hunk 7]
```diff
@@ -6,6 +6,7 @@ import { Component } from '@angular/core';
     <div className="row">
       <div className="example-spacing">
         <!-- Insert Example Here -->
+
```
Comment (mathsusan) [inline angular/src/docs/app/playground/playground.component.ts]: Take out added line

**Comment by mathsusan** [comment]
question about this .. can I put a custom component in a cell?  Not sure from all I read

**Comment by daanishnasir** [comment]
> question about this .. can I put a custom component in a cell? Not sure from all I read

@mathsusan You can, one of the examples I added has our avatar component in a custom cell

**Pull request merged**